### PR TITLE
vscode: install SVG icon

### PR DESCRIPTION
**Pull request by FnControlOption**
###### Description of changes

Installs SVG icon from https://code.visualstudio.com/brand

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/vscode/vscode.nix [hunk 1]
```diff
@@ -21,6 +21,11 @@ let
     aarch64-darwin = "1zhvbn6kmp99a5p0299256fm08y67rfzz03zygif7y20zrmr27ka";
     armv7l-linux = "0jn2li0j1cmk5m61ha6m4lskc80q1j7mfmgidc3x9x9yiv6yacr7";
   }.${system} or throwSystem;
+
+  icons = if stdenv.isDarwin || isInsiders then "" else fetchzip {
```
Comment (Mic92) [inline pkgs/applications/editors/vscode/vscode.nix:25]:
```suggestion
  icons = lib.optionalString (!stdenv.isDarwin && !isInsiders) fetchzip {
```

File: pkgs/applications/editors/vscode/vscode.nix [hunk 2]
```diff
@@ -42,6 +47,10 @@ in
 
     updateScript = ./update-vscode.sh;
 
+    postInstall = if stdenv.isDarwin || isInsiders then "" else ''
```
Comment (Mic92) [inline pkgs/applications/editors/vscode/vscode.nix:50]:
```suggestion
    postInstall = lib.optionalString (!stdenv.isDarwin && !isInsiders) ''
```

**Comment by superherointj** [comment]
Please fix merge conflict.

**Comment by drupol** [comment]
Hello,

While I liked the idea of having a SVG icon, I took the decision to close the PR by lack of feedback.

Feel free to reopen it if needed.

**Pull request closed without merging**